docs/rules/no-restricted-syntax.md

```diff
@@ -0,0 +1,51 @@
+# Disallow certain syntax (no-restricted-syntax)
+
+Some code bases prefer to not use certain features of JavaScript such as `FunctionExpression` or `WithStatement`. This rule warns on the passed node types.
```

Comment:
Okay, I'll try harder. That said if you have even a rough version in your mind, I'd welcome the help :)